### PR TITLE
Refactor clients for etcd-launcher

### PR DESCRIPTION
**Pull request by moelsayed**
**What this PR does / why we need it**:
Restructure etcd client code and remove some duplicated code.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/etcd-launcher/main.go [hunk 1]
```diff
@@ -242,12 +246,16 @@ func etcdCmd(config *config) []string {
...
+		fmt.Sprintf("--listen-peer-urls=https://%s:2380", config.podIP),
+		fmt.Sprintf("--initial-advertise-peer-urls=https://%s.etcd.%s.svc.cluster.local:2380", config.podName, config.namespace),
 		"--trusted-ca-file=/etc/etcd/pki/ca/ca.crt",
 		"--client-cert-auth",
 		"--cert-file=/etc/etcd/pki/tls/etcd-tls.crt",
 		"--key-file=/etc/etcd/pki/tls/etcd-tls.key",
+		"--peer-client-cert-auth",
+		"--peer-cert-file=/etc/etcd/pki/tls/etcd-tls.crt",
+		"--peer-key-file=/etc/etcd/pki/tls/etcd-tls.key",
+		"--peer-trusted-ca-file=/etc/etcd/pki/ca/ca.crt",
```
Comment (kron4eg) [inline cmd/etcd-launcher/main.go]: Two points:
* the hardcoded paths should be a part of [resources package](https://github.com/kubermatic/kubermatic/blob/master/pkg/resources/resources.go)
* https://github.com/kubermatic/kubermatic/blob/5fc026617f97b4268e7127777c857f58fd1317c3/pkg/resources/etcd/statefulset.go#L231-L235 points to the different tls key/cert that is intended to be used as peer certificate

Bonus:
Please align [pkg/resources/etcd/statefulset.go](https://github.com/kubermatic/kubermatic/blob/5fc026617f97b4268e7127777c857f58fd1317c3/pkg/resources/etcd/statefulset.go) in regard to existing and to-be-created constants to eliminate hardcoded paths

File: cmd/etcd-launcher/main.go [hunk 2]
```diff
@@ -260,31 +268,33 @@ func etcdCmd(config *config) []string {
...
+func (e *etcdCluster) getLocalClient() (*clientv3.Client, error) {
+	return e.getClientWithEndpoints([]string{e.endpoint()})
 }
 
 func (e *etcdCluster) getClientWithEndpoints(eps []string) (*clientv3.Client, error) {
 	var err error
+	tlsInfo := transport.TLSInfo{
+		CertFile:      "/etc/etcd/pki/tls/etcd-tls.crt",
+		KeyFile:       "/etc/etcd/pki/tls/etcd-tls.key",
+		TrustedCAFile: "/etc/etcd/pki/ca/ca.crt",
```
Comment (kron4eg) [inline cmd/etcd-launcher/main.go]: The same from previous comment about hardcoded paths

File: cmd/etcd-launcher/main.go [hunk 3]
```diff
@@ -260,31 +268,33 @@ func etcdCmd(config *config) []string {
...
+	}
+	tlsConfig, err := tlsInfo.ClientConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate client TLS config: %v", err)
+	}
+	tlsConfig.InsecureSkipVerify = true
```
Comment (kron4eg) [inline cmd/etcd-launcher/main.go]: ayay!!! Please don't.

**Review by kron4eg — CHANGES_REQUESTED** [review]

File: go.mod [hunk 4]
```diff
@@ -64,6 +64,7 @@ require (
 	github.com/stretchr/testify v1.6.1
 	github.com/urfave/cli v1.22.4
 	github.com/vmware/govmomi v0.22.2
+	go.etcd.io/etcd v0.5.0-alpha.5.0.20200716221620-18dfb9cca345
```
Comment (kron4eg) [inline go.mod]:
```suggestion
```
there is some error, it should be with /v3, like in the line below

Comment (moelsayed) [reply go.mod]: Unfortunately I don't think it's an error. it's a dependency introduced by /v3 itself. /v3 doesn't expose an interface to work with types like `member`, instead it just returns the internal type: https://github.com/kubermatic/kubermatic/blob/bef9d98285371cd8415a24ec7464e775e7f4da27/cmd/etcd-launcher/main.go#L355, I couldn't find a simple workaround for this.

Comment (moelsayed) [reply go.mod]: @kron4eg fixed, thanks man!

**Comment by kron4eg** [comment]
@moelsayed looks like you going to need to upgrade the golangci-lint (in here https://github.com/kubermatic/kubermatic/blob/7cf57ca9482d2d170a99f043b9216d8b5311e261/.prow.yaml#L159) in order to pass the linter step

**Review by kron4eg — APPROVED** [review]

**Pull request merged**